core/object/script_language.cpp

```diff
@@ -228,6 +233,16 @@ bool ScriptServer::is_global_class(const StringName &p_class) {
 	return global_classes.has(p_class);
 }
 
+StringName ScriptServer::get_global_class_name(const StringName &p_path) {
+	for (const KeyValue<StringName, GlobalScriptClass> &E : global_classes) {
+		if (E.value.path == p_path) {
+			return E.key;
+		}
+	}
```

Comment:
If the code is running in the editor, then it should use `EditorNode::get_editor_data().script_class_get_name(path)`) (something like that). EditorData maintains a reverse-lookup map to speed up these kinds of queries. Should be able to use `Engine::get_singleton()->is_editor_hint()` and the `#ifdef TOOLS_ENABLED` macro to help manipulate the contents of this method correctly.

Reply:
I'm unsure if adding this would make sense, It's my understanding the Editor should be separate from Core, but I believe I have seen these checks before in Core code, though I might be wrong, so I dont know lul.

Reply:
Yeah, I mentioned that they probably wouldn't want tool references in the core code in my edit. It's probably fine.